### ISSUE TITLE
feat: implement the fine-tuning procedure

### ISSUE THREAD

**Issue by longlephamtien** [longlephamtien/DL261-Assignment#70]
**Deliverable:** Configuration-driven fine-tuning procedure

### Acceptance criteria

- [ ] Layer freezing schedule, learning rates, scheduler, augmentation, and epochs are specified.
- [ ] The procedure is fully driven by configuration and reproducible.

### Planning

| Milestone | Priority | Start | Target | Estimate | Blocked by | Handbook |
| --- | --- | --- | --- | --- | --- | --- |
| A2-M3 Final | P0 | 2026-10-26 | 2026-10-31 | 3 | #66 | 20 |
